src/psyclone/gocean1p0.py

```diff
@@ -525,9 +533,17 @@ def _lower_bound(self):
             if go_kernels:
                 index_offset = go_kernels[0].index_offset
 
+            if self._loop_type == "inner":
+                stop = schedule.iloop_stop
+            else:
+                stop = schedule.jloop_stop
             if index_offset:
-                start = (self._bounds_lookup[index_offset][self.field_space]
-                         [self._iteration_space][self._loop_type]["start"])
+                # This strange line splitting was the only way I could find
+                # to avoid pep8 warnings: using [..._space]\ keeps on
+                # complaining about a white space
+                bounds = self._bounds_lookup[index_offset][self.field_space][
+                    self._iteration_space][self._loop_type]
+                start = bounds["start"].format(start='', stop=stop)
```

Comment:
It seems wrong/unnecessary to have to specify stop when getting a lower bound. Is it really necessary? My reading of the code is that the string stored under the "start" key won't have a `stop` format identifier?

Reply:
No, when you need a loop over a 1d halo region you will need both start and stop in upper and lower boundaries, e.g. for a (future) loop only for the eastern halo:
```
        self._bounds_lookup['offset_ne']['ct']['e_halo'] = \
            {'inner': {'start': '{stop}+1', 'stop': '{stop}+1'},
             'outer': {'start': "2", 'stop': "{stop}"}}
```
So in this case we actually only need an outer loop (across all rows, i.e. only j loop), the inner 'loop' is ```do i=istop+1, istop+1```, which on gen() will be replaced with ```i=istop+1``` (well, at least it will be like this once I have further pull requests in ;) ).

So we do need start and stop for both upper and lower boundaries.

Reply:
Ah yes - thanks for explaining.